src/utilities/alerter.ts

```diff
@@ -1,15 +1,22 @@
 import * as Bunyan from 'bunyan';
 import { sendEmail } from './email';
-import { IConfig } from '../types/IConfig';
+import { IConfig, IEmailConfig, AlertType } from '../types/IConfig';
 import { IDevice } from '../types/IDevice';
 import { arpscanDevicesAsync } from './scanner';
 
-interface IEmail {
+const defaultAlertMessages: { [alertType in AlertType]: string } = {
+	intruder: 'The fortress is in peril.',
+	departure: 'Goodbye.',
+	arrival: 'Welcome home.',
+	doorOpen: 'The door is open.',
+	knownDevice: 'A recognized device arrived.',
+}
+
+const alertTypes = Object.keys(defaultAlertMessages).reduce((map, key) => ({ ...map, [key]: key }), {}) as { [alertType in AlertType]: AlertType };

```

Comment:
hah, nice 👍

Reply:
The type on `defaultAlertMessages` requires that all `AlertTypes` exist as keys. So `alertTypes` here should always contain all possible types. `function triggerAlert(type: AlertType, ...` would have caused an error for passing strings that were not valid alert types, but if this is prefered, it is just as valid.